src/Microsoft.ML.Data/Scorers/FeatureContributionCalculatingTransformer.cs

```diff
@@ -0,0 +1,673 @@
+            }
+        }
+
+        /// <summary>
+        /// Holds the definition of the getters for the FeatureContribution column. It also contains the generic mapper that is used to score the Predictor.
+        /// </summary>
+        internal sealed class BindableMapper : ISchemaBindableMapper, ICanSaveModel, IPredictor
+        {
+            public sealed class Arguments : ScorerArgumentsBase
+            {
+                [Argument(ArgumentType.AtMostOnce, HelpText = "Number of top contributions", SortOrder = 1)]
+                public int Top = FeatureContributionCalculatingEstimator.Defaults.Top;
+
+                [Argument(ArgumentType.AtMostOnce, HelpText = "Number of bottom contributions", SortOrder = 2)]
+                public int Bottom = FeatureContributionCalculatingEstimator.Defaults.Bottom;
+
+                [Argument(ArgumentType.AtMostOnce, HelpText = "Whether or not output of Features contribution should be normalized", ShortName = "norm", SortOrder = 3)]
+                public bool Normalize = FeatureContributionCalculatingEstimator.Defaults.Normalize;
+
+                [Argument(ArgumentType.AtMostOnce, HelpText = "Whether or not output of Features contribution in string key-value format", ShortName = "str", SortOrder = 4)]
+                public bool Stringify = FeatureContributionCalculatingEstimator.Defaults.Stringify;
+
+                // REVIEW: the scorer currently ignores the 'suffix' argument from the base class. It should respect it.
+            }
+
+            private readonly int _topContributionsCount;
+            private readonly int _bottomContributionsCount;
+            private readonly bool _normalize;
+            private readonly IHostEnvironment _env;
+
+            public readonly IFeatureContributionMapper Predictor;
+            public readonly ISchemaBindableMapper GenericMapper;
+            public readonly bool Stringify;
+
+            internal const string MapperLoaderSignature = "WTFBindable";
+
+            private static VersionInfo GetVersionInfo()
+            {
+                return new VersionInfo(
+                    modelSignature: "WTF SCBI",
+                    verWrittenCur: 0x00010001, // Initial
+                    verReadableCur: 0x00010001,
+                    verWeCanReadBack: 0x00010001,
+                    loaderSignature: MapperLoaderSignature,
+                    loaderAssemblyName: typeof(FeatureContributionCalculatingTransformer).Assembly.FullName);
+            }
+
+            public PredictionKind PredictionKind => Predictor.PredictionKind;
+
+            public BindableMapper(IHostEnvironment env, IFeatureContributionMapper predictor, int topContributionsCount, int bottomContributionsCount, bool normalize, bool stringify)
+            {
+                Contracts.CheckValue(env, nameof(env));
+                _env = env;
+                _env.CheckValue(predictor, nameof(predictor));
+                if (topContributionsCount < 0)
```

Comment:
>< 0 [](start = 42, length = 3)

is = 0 acceptable?

The error says positive, but the check is for non-negative

Reply:
Sorry for me positive  = non negative, and includes 0. I will write non negative.

---
In reply to: [240788247](https://github.com/dotnet/machinelearning/pull/1847#discussion_r240788247) [](ancestors = 240788247)